llmfoundry/data/finetuning/tasks.py

```diff
@@ -332,6 +333,13 @@ def build_from_hf(
             preprocessing_fn = self.get_preprocessing_fn_from_str(
                 proto_preprocessing_fn, dataset_name)
 
+        signal_file_path = f'.node_{dist.get_node_rank()}_local_rank0_data_prep_completed'
+
+        if dist.get_local_rank() != 0:
+            log.debug('Waiting for local_rank 0 to finish data prep')
+            with dist.local_rank_zero_download_and_wait(signal_file_path):
+                dist.barrier()
```

Comment:
```suggestion
                pass
```
Instead, use one global barrier at end -- having ranks hit the same "effective" barrier at different points is very confusing